frontend/src/app/page.tsx

```diff
@@ -167,12 +168,25 @@ export default function Home() {
             {data.upcomingEvents.map((event: Event, index: number) => (
               <div key={`card-${event.name}`} className="overflow-hidden">
                 <div className="rounded-lg bg-gray-200 p-4 dark:bg-gray-700">
-                  <button
-                    className="mb-2 w-full cursor-pointer text-left text-lg font-semibold text-blue-400 hover:underline"
-                    onClick={() => setModalOpenIndex(index)}
-                  >
-                    <TruncatedText text={event.name} />
-                  </button>
+                  <div className="mb-2 flex items-center justify-between">
+                    <button
+                      className="cursor-pointer text-left text-lg font-semibold text-blue-400 hover:underline"
+                      onClick={() => setModalOpenIndex(index)}
+                    >
+                      <TruncatedText text={event.name} />
+                    </button>
+                    <CalendarButton
+                      event={{
+                        title: event.name,
+                        description: event.summary || '',
+                        location: event.suggestedLocation || '',
+                        startDate: event.startDate,
+                        endDate: event.endDate,
+                      }}
```

Comment:
@Kunal1522 Let's also pass in `event.url` and add it to the calendar event in the description.